src/cascadia/TerminalSettingsModel/GlobalAppSettings.cpp

```diff
@@ -88,13 +67,17 @@ GlobalAppSettings::GlobalAppSettings() :
 // - <none>
 void GlobalAppSettings::_FinalizeInheritance()
 {
-    // Globals only ever has 1 parent
-    FAIL_FAST_IF(_parents.size() > 1);
-    for (auto parent : _parents)
+    for (const auto& parent : _parents)
     {
         _actionMap->InsertParent(parent->_actionMap);
-        _keybindingsWarnings = std::move(parent->_keybindingsWarnings);
-        _colorSchemes = std::move(parent->_colorSchemes);
+        _keybindingsWarnings.insert(_keybindingsWarnings.end(), parent->_keybindingsWarnings.begin(), parent->_keybindingsWarnings.end());
+        for (const auto& [k, v] : parent->_colorSchemes)
+        {
+            if (!_colorSchemes.HasKey(k))
+            {
+                _colorSchemes.Insert(k, v);
+            }
+        }

```

Comment:
But how would that work when both the child and the parent(s) can have warnings & schemes simultaneously? After all I'm not using `LayerJson()` more than once per instance anymore.
I can clear the warnings & schemes in the parents after copying them though. What do you think?

Reply:
I'm of 3 thoughts:
1. **Ideal approach**: each layer has the warnings detected at that layer. When requested, we collect all of them and display them.
2. **Practical approach**: we don't care what each layer finds. We just care about the final one. So just std::move them down to the child and save the space.
3. **This approach (a middleground)**: each layer stores which warnings were detected here _and_ by our ancestors.

If storing all of the warnings more and more across each layer is expensive, I say we go with (2). Otherwise, I'm fine with this. (I don't know how expensive it can get so I'll leave that to you)

Reply:
It is unlikely to be expensive. Color Schemes are just reference-counted pointers, and warnings are just numbers!

Reply:
Originally I planned to finally fix layering for color schemes, but I'll do that in a follow-up PR.
Afterwards this code will be gone anyways. 🙂